src/components/Editor.vue

```diff
@@ -675,7 +675,10 @@ export default {
 				) {
 					this.dirty = state.dirty
 					if (this.dirty) {
+						window.addEventListener('beforeunload', this.saveBeforeUnload)
```

Comment:
I think this will be called multiple times per editing session. So we would end up with multiple listeners. So instead I propose we add this in the `mounted` hook and drop it in the `deleted` hook.

We only care about the case where the editor is not properly deleted here (directly closing the window for example). If the editor is closed properly we save already.

We will still need to make sure only to save if there are any changes in the undo history and `this.dirty` is true. But we can also check this in `saveBeforeUnload`.